src/relier/OAuthUtils.js

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ global browser
+ */
+
+const jose = require('node-jose');
+
+const KeyUtils = require('./KeyUtils');
+const util = require('./util');
+const fxaKeyUtils = new KeyUtils();
+
+const OAUTH_SERVER_URL = 'https://oauth.accounts.firefox.com/v1';
+
+/**
+ * OAuth methods to authenticate with Firefox Accounts and get scoped keys
+ * @module relier-OAuthUtils
+ * @example
+ * ```js
+ * const fxaKeysUtil = new fxaCryptoRelier.OAuthUtils();
+ *
+ * fxaKeysUtil.launchWebExtensionKeyFlow('YOUR_CLIENT_ID', {
+ *   redirectUri: browser.identity.getRedirectURL(),
+ *   scopes: ['profile', 'https://identity.mozilla.org/apps/lockbox'],
+ * }).then((loginDetails) => {
+ *   const key = loginDetails.keys['https://identity.mozilla.org/apps/lockbox'];
+ *   const credentials = {
+ *     access_token: loginDetails.access_token,
+ *     refresh_token: loginDetails.refresh_token,
+ *     key
+ * };
+ * ```
+ */
+class OAuthUtils {
+  /**
+   * @constructor
+   * @param {object} [options]
+   * @param {string} [options.oauthServer] - Custom Firefox Accounts OAuth server
+   */
+  constructor(options = {}) {
+    this.oauthServer = options.oauthServer || OAUTH_SERVER_URL;
+  }
+  /**
+   * @method launchWebExtensionKeyFlow
+   * @desc Used to launch the Firefox Accounts scope key login flow in WebExtensions
+   * @param {string} clientId - FxA relier client id
+   * @param {object} [options={}]
+   * @param {array} [options.scopes=[]] - Requested OAuth scopes
+   * @param {object} [options.browserApi=browser] - Custom browser API override
+   * @param {function} [options.getBearerTokenRequest=getBearerTokenRequest] -
+   *   Custom getBearerTokenRequest function override
+   * @returns {Promise}
+   */
+  launchWebExtensionKeyFlow(clientId, options = {}) {
+    if (! clientId) {
+      throw new Error('clientId required');
+    }
+
+    const browserApi = options.browserApi || browser;
+    const getBearerTokenRequest = options.getBearerTokenRequest || this._getBearerTokenRequest;
+    const SCOPES = options.scopes || [];
+
+    const state = util.createRandomString(16);
+    const codeVerifier = util.createRandomString(32);
+    const queryParams = {
+      access_type: 'offline', // eslint-disable-line camelcase
+      client_id: clientId, // eslint-disable-line camelcase
+      redirect_uri: options.redirectUri, // eslint-disable-line camelcase
+      scope: SCOPES.join(' '),
+      state: state
+    };
+
+    let authUrl = `${this.oauthServer}/authorization`;
```

Comment:
Ideally, we would discover this URL by dynamic lookup of `/.well-known/openid-configuration` on the server, since the fact that it's exposed at `/authorization` is an internal implementation detail of FxA.

Reply:
But see https://github.com/mozilla/fxa-content-server/issues/5453 for a potential reason why we can't.